lib/commands/recordscreen.js

```diff
@@ -0,0 +1,277 @@
+ *                              Only works if `remotePath` is provided.
+ * @property {?string} videoType - The format of the screen capture to be recorded.
+ *                                 Available formats: "h264", "mp4" or "fmp4". Default is "mp4".
+ *                                 Only works for Simulator.
+ * @property {?string} videoQuality - The video encoding quality (low, medium, high, photo - defaults to high).
+ *                                    Only works for real devices.
+ * @property {?boolean} forceRestart - Whether to try to catch and upload/return the currently running screen recording
+ *                                     (`false`, the default setting) or ignore the result of it and start a new recording
+ *                                     immediately.
+ * @property {?string|number} timeLimit - The maximum recording time, in seconds.
+ *                                        The default value is 180, the maximum value is 600 (10 minutes).
+ */
+
+/**
+ * Record the display of devices running iOS Simulator since Xcode 8.3 or real devices since iOS 8
+ * (xrecord utility is required: https://github.com/WPO-Foundation/xrecord).
+ * It records screen activity to an MPEG-4 file. Audio is not recorded with the video file.
+ * If screen recording has been already started then the command will stop it forcefully and start a new one.
+ * The previously recorded video file will be deleted.
+ *
+ * @param {?StartRecordingOptions} options - The available options.
+ * @returns {string} Base64-encoded content of the recorded media file if
+ *                   any screen recording is currently running or an empty string.
+ * @throws {Error} If screen recording has failed to start.
+ */
+commands.startRecordingScreen = async function (options = {}) {
+  const {videoType, timeLimit, videoQuality, forceRestart} = options;
+
+  let result = '';
+  if (!forceRestart) {
+    result = await this.stopRecordingScreen(options);
+  }
+  const pgrepPattern = this.isRealDevice() ? REAL_DEVICE_PGREP_PATTERN : SIMULATOR_PGREP_PATTERN;
+  const pid = await getPidUsingPattern(pgrepPattern(this.opts.device.udid));
+  if (!_.isEmpty(pid)) {
+    try {
+      await exec('kill', [pid]);
+    } catch (err) {
+      log.errorAndThrow(`Unable to stop screen recording process: ${err.message}`);
+    }
+  }
+  if (!_.isEmpty(this._recentScreenRecordingPath)) {
+    await fs.rimraf(this._recentScreenRecordingPath);
+    this._recentScreenRecordingPath = null;
+  }
+
+  const localPath = temp.path({prefix: 'appium', suffix: DEFAULT_EXT});
+
+  let binaryName;
+  const args = [];
+  if (this.isRealDevice()) {
+    binaryName = REAL_DEVICE_BINARY;
+    if (!await fs.which(binaryName)) {
+      log.errorAndThrow(`'${binaryName}' binary is not found in PATH. Make sure it is present on the system. ` +
+                        `Check https://github.com/WPO-Foundation/xrecord for more details.`);
+    }
+    args.push('--quicktime',
```

Comment:
Both real device and simulator case have base arguments, so pushing is not necessary, in the same way as `binaryName` is handled.
```js
let binaryName;
let args;
if (this.isRealDevice()) {
  binaryName = REAL_DEVICE_BINARY;
  // ...
  args = {
    '--quicktime',
    `--id="${this.opts.device.udid}"`,
    `--out="${localPath}"`,
    `--force`,
  };
  // ...
} else {
  binaryName = SIMULATOR_BINARY;
  args = {
    'simctl', 
    'io', 
    this.opts.device.udid, 
    'recordVideo',
  };
  // ...
}
```

Reply:
for me it is just a matter of style. You now how I like constants :)

Reply:
But it is logically _not_ a constant. :)